giotto/homology/preprocessing.py

```diff
@@ -0,0 +1,160 @@
+        using all processors.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> X = np.random.rand(10, 7)
+    >>> space_extraction = LocalNeighborhood(dist_percentage=0.3, k_min=2,
+    ...                                      k_max=5)
+    >>> Xt = space_extraction.fit_transform(X)
+    >>> print(Xt.shape)
+    (10, 2, 7)
+
+    """
+    _hyperparameters = {'dist_percentage': [float, (0, 1)],
+                        'k_min': [int, (1, np.inf)],
+                        'k_max': [int, (1, np.inf)]
+                        }
+
+    def __init__(self, dist_percentage=0.05, k_min=10, k_max=100,
+                 metric='euclidean', n_jobs=None):
+        self.dist_percentage = dist_percentage
+        self.k_min = k_min
+        self.k_max = k_max
+        self.metric = metric
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged.
+        This method is there to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Input data.
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        validate_params(self.get_params(), self._hyperparameters)
+        check_array(X)
+        _check_k_min_max(self.k_min, self.k_max)
+
+        self._is_fitted = True
+
+        return self
+
+    def transform(self, X, y=None):
+        """For each entry in the input data array ``X``, returns the point
+        cloud surrounding it.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Input data.
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        Xt : ndarray, shape (n_samples, n_points, n_features)
+            Array containing the (local) point clouds associated with each
+            entry of ``X``.
+
+        """
+        check_is_fitted(self, attributes=["_is_fitted"])
+        check_array(X)
+
+        distance_matrix = squareform(pdist(X, self.metric))
+
+        local_point_clouds = Parallel(n_jobs=self.n_jobs)(
+            delayed(self._extract_point_clouds)(X, distance_matrix, i)
+            for i in range(len(X))
+        )
+        max_n_points = np.max([x.shape[0] for x in local_point_clouds])
+
+        Xt_dims = list(X.shape)
+        Xt_dims.insert(1, max_n_points)
+        Xt = np.empty(Xt_dims)
+
+        for i, point_cloud in enumerate(local_point_clouds):
+            Xt[i, : len(point_cloud[0])] = point_cloud[0]
+            Xt[i, len(point_cloud[0]):] = point_cloud[0][-1]
+
+        return Xt
+
+    def _extract_point_clouds(self, X, matrix_distances, ind_x):
```

Comment:
Favour naming consistency with usage in transform (matrix_distances becomes distance_matrix there)